pkgs/os-specific/darwin/swift-corelibs/corefoundation.nix

```diff
@@ -21,29 +21,38 @@ stdenv.mkDerivation {
     sha256 = "17kpql0f27xxz4jjw84vpas5f5sn4vdqwv10g151rc3rswbwln1z";
   };
 
-  nativeBuildInputs = [ ninja python3 ];
-  buildInputs = [ curl libxml2 objc4 ICU ];
-
-  patches = [ ./0001-Add-missing-TARGET_OS_-defines.patch ];
+  nativeBuildInputs = [ cmake ninja pkg-config python3 ];
+  buildInputs = [ (lib.getDev launchd) libdispatch libxml2 objc4 icu ];
+
+  patches = [
+    ./0001-Add-missing-TARGET_OS_-defines.patch
+    # CFMessagePort.h uses `bootstrap_check_in` without declaring it, which is defined in the launchd headers.
+    ./0002-Add-missing-launchd-header.patch
+    # CFURLComponents fails to build with clang 16 due to an invalid pointer conversion. This is fixed upstream.
+    ./0003-Fix-incompatible-pointer-conversion.patch
+    # Fix `CMakeLists.txt` to allow it to be used instead of `build.py` to build on Darwin.
+    ./0004-Fix-Darwin-cmake-build.patch
+    # Install CF framework in `$out/Library/Frameworks` instead of `$out/System/Frameworks`.
+    ./0005-Fix-framework-installation-path.patch
+    # Build a framework that matches the contents of the system CoreFoundation. This patch adds
+    # versioning and drops the prefix and suffix, so the dynamic library is named `CoreFoundation`
+    # instead of `libCoreFoundation.dylib`.
+    ./0006-System-CF-framework-compatibility.patch
+    # Link against the nixpkgs ICU instead of using Apple’s vendored version.
+    ./0007-Use-nixpkgs-icu.patch
+    # Don’t link against libcurl. This breaks a cycle between CF and curl, which depends on CF via
+    # and the SystemConfiguration framework to support NAT64.
+    # This is safe because the symbols provided in CFURLSessionInterface are not provided by the
+    # system CoreFoundation. They are meant to be used by the implementation of `NSURLSession` in
+    # Swift Foundation, which is not built (nor should it be due to incompatibilities) on Darwin.
```

Comment:
I don't think the Swift Foundation is ready to replace the macOS one yet. (And in general it's not clear to me that trying to build the OS-shipped SDKs from source is a good idea when it's impossible to do so for all of them; it's caused plenty of problems even with CF.)

Reply:
It’s not that it can’t. It’s that it _shouldn’t_ be built. The Foundation provided by swift-corelibs-foundation is not fully compatible with the system Foundation. APIs are missing, work differently, or are buggy. It’s a major pain point for Swift on Linux. Darwin maintenance already has enough challenges. I would not want to add to them by using an incompatible Foundation implementation (especially if it’s done conditionally like CF is today).

However, it’s possible that we may be able to build its replacement in the future. The Swift project announced a while back that [Foundation is being rewritten in pure swift][rewrite], and Apple plans to ship it as the Foundation implementation in macOS. To maintain compatibility, Apple will include additional legacy APIs in its implementation beyond what Swift rewrite provides. If the source is released, we can build that to produce a compatible Foundation.

The [Swift Foundation][foundation-repo] is here. If it would help clarify the comment, I can refer to “swift-corelibs-foundation” instead to distinguish the two versions.

[rewrite]: https://www.swift.org/blog/future-of-foundation/
[foundation-repo]: https://github.com/apple/swift-foundation

Reply:
> The [Swift Foundation](https://github.com/apple/swift-foundation) is here. If it would help clarify the comment, I can refer to “swift-corelibs-foundation” instead to distinguish the two versions.

I think this would be helpful; I assumed "Swift Foundation" was referring to that rewrite (hence my comment).

Reply:
I’ve updated the comment in the latest push to use swift-corelibs-foundation instead and notes that it’s not built because it’s not fully compatible with the system Foundation used on Darwin.